source/common/upstream/outlier_detection_impl.cc

```diff
@@ -471,8 +475,28 @@ void DetectorImpl::ejectHost(HostSharedPtr host,
           runtime_.snapshot().getInteger(BaseEjectionTimeMsRuntime, config_.baseEjectionTimeMs()));
       const std::chrono::milliseconds max_eject_time = std::chrono::milliseconds(
           runtime_.snapshot().getInteger(MaxEjectionTimeMsRuntime, config_.maxEjectionTimeMs()));
-      if ((host_monitors_[host]->ejectTimeBackoff() * base_eject_time) <
-          (max_eject_time + base_eject_time)) {
+      const std::chrono::milliseconds max_eject_time_jitter =
+          std::chrono::milliseconds(runtime_.snapshot().getInteger(
+              MaxEjectionTimeJitterMsRuntime, config_.maxEjectionTimeJitterMs()));
+
+      // Generate random jitter so that not all hosts uneject at the same time,
+      // which could possibly generate a connection storm
+
+      // std::chrono::milliseconds receives a signed integer as input, so we ensure that our random
+      // random value is in the range [0, std::numeric_limits<long>::max())
+      long random_long =
```

Comment:
nit: const

Not sure I understand why the result is casted from uint64 to long.
Can the computation be done as follows:
```
const std::chrono::milliseconds jitter = 
    std::chrono::milliseconds(random_generator_() % (max_eject_time_jitter + std::chrono::milliseconds(1));
```

Reply:
std::chrono::milliseconds receives an argument that is a signed integer that is at least 45 bits long (https://en.cppreference.com/w/cpp/chrono/duration). If you give it a uint64 as a parameter it might return a negative value.

Reply:
Maybe I'm missing something here, but if the modulo operation (which should be non-negative, and keep the result in the milliseconds range) is applied, and then the cast to milliseconds, it should stay non-negative.

Reply:
Fixed